app/services/DynamoBanditData.scala

```diff
@@ -108,11 +110,11 @@ class DynamoBanditData(stage: String, client: DynamoDbClient) extends StrictLogg
               logger.error(s"Failed to decode item from Dynamo: ${error.getMessage}")
               None
           }
-          .toList
+          .toArray
```

Comment:
Switched to using an array internally here because we now need to get subarrays (see the `.slice` above), which would be very inefficient with a List